ci/supervisor/config/config.go

```diff
@@ -0,0 +1,329 @@
+	d.CodeCoverage = structuredData.CodeCoverage
+	d.ProwPlugins = make(map[string]ProwPlugin, 0)
+	return parsePlugins(d.ProwPlugins, structuredData.ProwPlugins)
+}
+
+// ParseRepoConfig unmarshals a repository config YAML
+func ParseRepoConfig(content []byte) (RepoConfig, error) {
+	r := RepoConfig{}
+	r.PresubmitTasks = make(map[string]PresubmitTask, 0)
+	r.PeriodicTasks = make(map[string]PeriodicTask, 0)
+	r.PostsubmitTasks = make(map[string]PostsubmitTask, 0)
+
+	// Extract the raw YAML
+	var rawConfig map[string]interface{}
+	if err := yaml.Unmarshal(content, &rawConfig); err != nil {
+		return RepoConfig{}, err
+	}
+
+	// Unmarshal the defaults section
+	if d, err := yaml.Marshal(rawConfig["defaults"]); err == nil {
+		if err := yaml.Unmarshal(d, &r.Defaults); err != nil {
+			return RepoConfig{}, err
+		}
+	} else {
+		return RepoConfig{}, err
+	}
+
+	// Unmarshal the presubmit-tasks section
+	if value, ok := rawConfig["presubmit-tasks"].(yamlMap); ok {
+		for a := range value {
+			taskName, _ := a.(string)
+			// Setting can be simply the command to run
+			if v, ok := value[taskName].(string); ok {
+				r.PresubmitTasks[taskName] = PresubmitTask{TaskParameters: Task{Command: v}}
+				// Otherwise it's the complete set of parameters
+			} else {
+				if d, err := yaml.Marshal(value[taskName]); err == nil {
+					pt := PresubmitTask{}
+					err = yaml.Unmarshal(d, &pt)
+					err = yaml.Unmarshal(d, &pt.TaskParameters)
+					r.PresubmitTasks[taskName] = pt
+				} else {
+					return RepoConfig{}, err
+				}
+			}
+		}
+	}
+
+	// Unmarshal the periodic-tasks section
+	if value, ok := rawConfig["periodic-tasks"].(yamlMap); ok {
+		for a := range value {
+			taskName, _ := a.(string)
+			// Expect the complete set of parameters
+			if d, err := yaml.Marshal(value[taskName]); err == nil {
+				pt := PeriodicTask{}
+				err = yaml.Unmarshal(d, &pt)
+				err = yaml.Unmarshal(d, &pt.TaskParameters)
+				r.PeriodicTasks[taskName] = pt
+			} else {
+				return RepoConfig{}, err
+			}
+		}
+	}
+
+	// Unmarshal the postsubmit-tasks section
+	if value, ok := rawConfig["postsubmit-tasks"].(yamlMap); ok {
+		for a := range value {
+			taskName, _ := a.(string)
+			// Setting can be simply the command to run
+			if v, ok := value[taskName].(string); ok {
+				r.PostsubmitTasks[taskName] = PostsubmitTask{TaskParameters: Task{Command: v}}
+				// Otherwise it's the complete set of parameters
+			} else {
+				if d, err := yaml.Marshal(value[taskName]); err == nil {
+					pt := PostsubmitTask{}
+					err = yaml.Unmarshal(d, &pt)
+					err = yaml.Unmarshal(d, &pt.TaskParameters)
+					r.PostsubmitTasks[taskName] = pt
+				} else {
+					return RepoConfig{}, err
+				}
+			}
+		}
+	}
+
+	return r, nil
+}
+
+// UnmarshalYAML unmarshals a SupervisorDefaults struct, correctly unmarshalling the Prow plugin section
+func (sd *SupervisorDefaults) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	// Expected fields
+	var structuredData struct {
```

Comment:
Why not just use sd itself? would it be easier if this method becomes a global helper method like this:
```
func UnmarshalYAMLToSuperVisorDefaults(unmarshal func(interface{}) error) (*SupervisorDefaults, error) {
```